std/file.d

```diff
@@ -3200,6 +3200,219 @@ version(Posix) unittest //issue 11434
     assert(getAttributes(t2) == octal!100767);
 }
 
+/***************************************************
+Copy file or directory $(D from) to $(D to). When a directory is copied, its
+contents are copied as well, recursively.
+
+File timestamps are preserved, except for symbolic links.
+File attributes are preserved, if $(D preserve) equals $(D PreserveAttributes.yes).
+On Windows only $(D PreserveAttributes.yes) (the default on Windows) is supported.
+
+Existing target files are overwritten. Existing target directories are written
+into.
+
+With symbolic links, the links themselves are copied. That is, for every
+symbolic link in $(D from), a new link with the same target path is created in
+$(D to). Relative target paths are not changed. They may be invalidated by the
+copying, when they point outside of the copied directory tree.
+
+Special files (e.g. device files, FIFOs) cannot be copied with $(D copyRecurse).
+
+Params:
+    from = string or range of characters representing the existing
+        file/directory name
+    to = string or range of characters representing the target file/directory
+        name
+    preserve = whether to preserve the file attributes
+
+Throws: $(D FileException) on error.
+ */
+void copyRecurse(string from, string to,
+    PreserveAttributes preserve = preserveAttributesDefault)
+{
+    // Remove any trailing directory separators.
+    from = buildNormalizedPath(from);
+    to = buildNormalizedPath(to);
+
+    immutable attrs = getLinkAttributes(from);
+
+    if (attrIsFile(attrs)) copy(from, to, preserve);
```

Comment:
I'd very much like to see this work where `to` is a directory (and the documentation you wrote implies that it should). `copy` can't handle that. It insists that you list the file name at the end of the target rather than just a directory (which I find _very_ annoying). You can see a discussion on it in this bug report: https://issues.dlang.org/show_bug.cgi?id=3862. The conclusion of that was that we decided that it would probably be better to leave `copy` as-is and create a new function which copied similar to what you've done here (much as I think that `copy` is it stands is pretty useless). I still need to look over exactly how this implementation behaves, but what I'd like to see is a function that roughly does what `cp` does.